buildengine/engine.go

```diff
@@ -82,10 +83,11 @@ func WithListener(listener Listener) Option {
 // pull in missing schemas.
 //
 // "dirs" are directories to scan for local modules.
-func New(ctx context.Context, client ftlv1connect.ControllerServiceClient, moduleDirs []string, externalDirs []string, options ...Option) (*Engine, error) {
+func New(ctx context.Context, client ftlv1connect.ControllerServiceClient, projConfig *projectconfig.Config, moduleDirs []string, externalDirs []string, options ...Option) (*Engine, error) {
```

Comment:
`projectconfig.Config` is not the appropriate type for this use. It is the raw configuration struct used to unmarshal TOML files, so it shouldn't be used outside that scope. `ModuleContext` OTOH is the appropriate type to use here, but IIRC there's no mechanism to get one prior to build. This needs some more thought, but my gut instinct is telling me we need an abstraction to retrieve `ModuleConfig`s from a `projectconfig.Config`.

Reply:
I think if implementing the above is too much work we should abandon this ticket for now as it's not super high priority. Perhaps timebox it.

Reply:
Ahhhhh got it, that explains a _lot_. I'll table this for now and pick up one of the `next` issues. We've talked a few times lately about possibly refactoring the interfaces for the TOML files, so it probably makes the most sense to come up with a holistic design for that that includes this use case.

Reply:
Just happened to see https://github.com/TBD54566975/ftl/issues/919 in `next` :) Just to make sure I'm reading correctly, is that issue basically the blocker to this one?